test/models/conversation_test.rb

```diff
@@ -47,4 +47,34 @@ class ConversationTest < ActiveSupport::TestCase
       conversation.reload
     end
   end
+
+  test "#grouped_by_increasing_time_interval_for_user" do
+    Timecop.freeze do
```

Comment:
There's a really tiny probability that this test happing when the clock rolls over a second, which causes a roll over of a week, month, or year.

Doing that would cause this code to break, which is perhaps a remote enough possibility that Timecop isn't warranted. On the other hand, I've used timecop for a _long time_ and it has show itself to be stable and well maintained for many years.

Open to suggestions!

Reply:
Agreed, timecop is very nice and flappy tests suck. :)